README.md

```diff
@@ -1,223 +1,67 @@
+- execute commands with a functional API.
+- receive message and other notifications via STM queue.
+- automatically send keep-alive commands.
 
-To specify a different file path for the chat database use `-d` command line option:
+### SMP agent
 
-```shell
-$ dog-food -d my-chat.db
-```
+[SMP agent library](./src/Simplex/Messaging/Agent.hs) can be used to run SMP agent as part of another application and to communicate with the agent via STM queues, without serializing and parsing commands and responses.
 
-If you deployed your own SMP server you can set client to use it via `-s` option:
+Haskell type [ACommand](./src/Simplex/Messaging/Agent/Transmission.hs) represents SMP agent protocol to communicate via STM queues.
 
-```shell
-$ dog-food -s smp.example.com:5223
-```
+See [simplex-chat](https://github.com/simplex-chat/simplex-chat) terminal UI for the example of integrating SMP agent into another application.
 
-You can still talk to people using default or any other server, it only affects the location of the message queue when you initiate the connection (and the reply queue can be on another server, as set by the other party's client).
+[SMP agent executable](./apps/smp-agent/Main.hs) can be used to run a standalone SMP agent process that implements plaintext [SMP agent protocol](./protocol/agent-protocol.md) via TCP port 5224, so it can be used via telnet. It can be deployed in private networks to share access to the connections between multiple applications and services.
 
-Run `dog-food --help` to see all available options.
+## Using SMP server and SMP agent
 
-### Using chat client
+You can either run SMP server locally or try local SMP agent with the deployed demo server `smp1.simplex.im:5223#pLdiGvm0jD1CMblnov6Edd/391OrYsShw+RgdfR0ChA=`
 
-Once chat client is started, use `/add <name1>` to create a new connection and generate an invitation to send to your contact via any other communication channel (`<name1>` - is any name you want to use for that contact).
+It's the easiest to try SMP agent via a prototype [simplex-chat](https://github.com/simplex-chat/simplex-chat) terminal UI.
 
-Invitation has format `smp::<server>::<queue_id>::<public_key_for_this_queue_only>` - this needs to be shared with another party, via any other chat. It can only be used once - even if this is intercepted, the attacker would not be able to use it to send you the messages via this queue once your contact confirms that the connection is established.
+## SMP server design
 
-The party that received the invitation should use `/accept <name2> <invitation>` to accept the connection (`<name2>` is any name that the accepting party wants to use for you).
+![SMP server design](./design/server.svg)
 
-For example, if Alice and Bob want to chat, with Alice initiating, Alice would use [in her chat client]:
+## SMP agent design
 
-```
-/add bob
-```
+![SMP agent design](./design/agent2.svg)
```

Comment:
there's a `Q: can multiple threads use it` not on sqlite block - shall we remove it? I mean we know they can.. with caveats